src/Game/Ride/SurfRay.cpp

```diff
@@ -0,0 +1,1038 @@
+}
+
+void SurfRay::attackSensor(HitSensor* pSender, HitSensor* pReceiver) {
+    MR::tryGetItem(pSender, pReceiver);
+}
+
+bool SurfRay::receiveOtherMsg(u32 msg, HitSensor* pSender, HitSensor* pReceiver) {
+    if (MR::isMsgAutoRushBegin(msg)) {
+        if (MR::getPlayerVelocity()->y > 0.0f) { // inline MarioAccess::getVelocity ??
+            return false;
+        }
+        else {
+            mRider = MR::getSensorHost(pSender);
+            MR::startStarPointerMode1PInvalid2PValid(this);
+            MR::invalidateClipping(this);
+            mRayTilt = 0.0f;
+            if (MR::isValidSwitchA(this)) {
+                MR::onSwitchA(this);
+            }
+            MR::startBckNoInterpole(this, "Wait");
+            MR::startBckPlayer("SurfRideLoop", (s32)0); // MarioAccess::changeAnimationE() ??
+            MR::calcAnimDirect(this);
+            mActorJointCtrl->resetDynamicCtrl();
+            if (mInTutorial) {
+                setNerve(&NrvSurfRay::SurfRayNrvTutorial::sInstance);
+            }
+            else {
+                setNerve(&NrvSurfRay::SurfRayNrvReady::sInstance);
+            }
+            return true;
+        }
+    }
+    else if (MR::isMsgUpdateBaseMtx(msg)) {
+        updateRiderMtx();
+        return true;
+    }
+    else if (MR::isMsgRushCancel(msg)) {
+        mRider = nullptr;
+        setNerve(&NrvSurfRay::SurfRayNrvRideFree::sInstance);
+        return true;
+    }
+    else if (MR::isMsgTutorialStart(msg)) {
+        MR::setPlayerPos(mPosition);
+        mInTutorial = true;
+        return true;
+    }
+    else if (MR::isMsgTutorialNext(msg)) {
+        mLectureIdx++;
+        setNerve(&NrvSurfRay::SurfRayNrvTutorial::sInstance);
+        return true;
+    }
+    else if (MR::isMsgTutorialPrev(msg)) {
+        mLectureIdx--;
+        setNerve(&NrvSurfRay::SurfRayNrvTutorial::sInstance);
+        return true;
+    }
+    else if (MR::isMsgTutorialPass(msg)) {
+        mInTutorial = false;
+        setNerve(&NrvSurfRay::SurfRayNrvRideFree::sInstance);
+        return true;
+    }
+    else if (MR::isMsgTutorialOmit(msg)) {
+        mInTutorialArea = false;
+        MR::resetPosition(this, "スタート位置（サーフィン）");
+        resetAllInfo();
+        MR::setPlayerPos(mPosition);
+        MR::startBckPlayer("SurfRide", (s32)0); // MarioAccess::changeAnimationE() ??
+        MR::resetCameraMan();
+        return true;
+    }
+    else if (MR::isMsgRaceReady(msg)) {
+        setNerve(&NrvSurfRay::SurfRayNrvReady::sInstance);
+        return true;
+    }
+    else if (MR::isMsgRaceStart(msg)) {
+        setNerve(&NrvSurfRay::SurfRayNrvRideFree::sInstance);
+        return true;
+    }
+    else if (MR::isMsgRaceReset(msg)) {
+        MR::endBindAndPlayerWait(this);
+        resetAllInfo();
+        setNerve(&NrvSurfRay::SurfRayNrvWait::sInstance);
+        MR::endStarPointerMode(this);
+        kill();
+        return true;
+    }
+    else {
+        return false;
+    }
```

Comment:
Prefer guard clauses over chained `if`-`else if`-`else` statements in situations like this where each block contains a `return` statement.